src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java

```diff
@@ -0,0 +1,131 @@
+package mate.academy.internetshop.dao.jdbc;
+
+import mate.academy.internetshop.dao.BucketDao;
+import mate.academy.internetshop.dao.ItemDao;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.lib.Inject;
+import mate.academy.internetshop.model.Bucket;
+import mate.academy.internetshop.model.Item;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.log4j.Logger;
+
+@Dao
+public class BucketDaoJdbcImpl extends AbstractDao<Bucket> implements BucketDao {
+    private static Logger logger = Logger.getLogger(BucketDaoJdbcImpl.class);
+    @Inject
+    private static ItemDao itemDao;
+
+    public BucketDaoJdbcImpl(Connection connection) {
+        super(connection);
+    }
+
+    @Override
+    public Bucket add(Bucket bucket) {
```

Comment:
better use `try with resources` in all methods
try to divide this method into two different (add() and addItemToBucket())